execution.tf

```diff
@@ -0,0 +1,60 @@
+resource "aws_iam_role" "execution" {

```

Comment:
This role needs to be attached to the Beanstalk environment.
This is done by creating an instance profile and attaching it as a setting.
1. Create an instance profile
```
resource "aws_iam_instance_profile" "execution" {
  name = local.resource_name
  role = aws_iam_role.execution.name
}
```
2. Add to `local.basic_settings`
```
  basic_settings = [
    ...
    {
      namespace = "aws:autoscaling:launchconfiguration"
      name      = "IamInstanceProfile"
      value     = aws_iam_instance_profile.execution.name
    }
  ]
```

Reply:
Right, good point. We already have an Instance Profile that we're using for Beanstalk and I don't think we can have more than one, so I'm going to add the permissions to the existing role we're creating instead.

Reply:
I moved some of this around, so let me know if the structure makes sense or if we should organize it differently.